shortfin/tests/apps/llm/cpu_llm_server_test.py

```diff
@@ -0,0 +1,156 @@
+
+        # Set up tokenizer if it doesn't exist
+        tokenizer_path = os.path.join(tmp_dir, "tokenizer.json")
+        if not os.path.exists(tokenizer_path):
+            tokenizer = AutoTokenizer.from_pretrained(
+                "openlm-research/open_llama_3b_v2"
+            )
+            tokenizer.save_pretrained(tmp_dir)
+
+        # Export model if it doesn't exist
+        mlir_path = os.path.join(tmp_dir, "model.mlir")
+        config_path = os.path.join(tmp_dir, "config.json")
+        if not os.path.exists(mlir_path) or not os.path.exists(config_path):
+            bs_string = ",".join(map(str, BATCH_SIZES))
+            subprocess.run(
+                [
+                    "python",
+                    "-m",
+                    "sharktank.examples.export_paged_llm_v1",
+                    f"--gguf-file={model_path}",
+                    f"--output-mlir={mlir_path}",
+                    f"--output-config={config_path}",
+                    f"--bs={bs_string}",
+                ],
+                check=True,
+            )
+        # Compile model if it doesn't exist
+        vmfb_path = os.path.join(tmp_dir, "model.vmfb")
+        if not os.path.exists(vmfb_path):
+            subprocess.run(
+                [
+                    "iree-compile",
+                    mlir_path,
+                    "-o",
+                    vmfb_path,
+                ]
+                + settings["device_flags"],
+                check=True,
+            )
+        # Write config if it doesn't exist
+        edited_config_path = os.path.join(tmp_dir, "edited_config.json")
+        if not os.path.exists(edited_config_path):
+            config = {
+                "module_name": "module",
+                "module_abi_version": 1,
+                "max_seq_len": 2048,
+                "attn_head_count": 32,
+                "attn_head_dim": 100,
+                "prefill_batch_sizes": BATCH_SIZES,
+                "decode_batch_sizes": BATCH_SIZES,
+                "transformer_block_count": 26,
+                "paged_kv_cache": {"block_seq_stride": 16, "device_block_count": 256},
+            }
+            with open(edited_config_path, "w") as f:
+                json.dump(config, f)
+        yield tmp_dir
+    finally:
+        shutil.rmtree(tmp_dir)
+
+
+@pytest.fixture(scope="module")
+def llm_server(model_test_dir):
+    # Start the server
+    server_process = subprocess.Popen(
+        [
+            "python",
+            "-m",
+            "shortfin_apps.llm.server",
+            f"--tokenizer={os.path.join(model_test_dir, 'tokenizer.json')}",
+            f"--model_config={os.path.join(model_test_dir, 'edited_config.json')}",
+            f"--vmfb={os.path.join(model_test_dir, 'model.vmfb')}",
+            f"--parameters={os.path.join(model_test_dir, 'open-llama-3b-v2-f16.gguf')}",
+            f"--device={settings['device']}",
+        ]
+    )
+    # Wait for server to start
+    time.sleep(2)
+    yield server_process
```

Comment:
Sketchy timeout. Might be flaky on CI. Is there a more reliable way to wait for startup?

Reply:
Changed to using `/health` route of server to verify that it's started properly